src/erlfmt_format.erl

```diff
@@ -307,14 +307,23 @@ binary_op_to_algebra('/', _Meta, {atom, _, _} = Left, {integer, _, _} = Right) -
 binary_op_to_algebra('|', Meta, Left, Right) ->
     %% | does not increase indentation and breaks as a whole
     union_to_algebra(Meta, Left, Right);
-binary_op_to_algebra(Op, Meta0, Left, Right) ->
+binary_op_to_algebra(Op, Meta0, Left0, Right0) ->
     %% don't print parens and comments twice - expr_to_algebra took care of it already
     Meta = erlfmt_scan:delete_annos([parens, pre_comments, post_comments], Meta0),
+    {op, _Meta, Op, Left, Right} = rewrite_assoc(Op, Meta, Left0, Right0),
     binary_op_to_algebra(Op, Meta, Left, Right, ?INDENT).
 
+rewrite_assoc(Op, MetaA, A, {op, MetaBC, Op, B, C} = BC) ->
+    case erlfmt_scan:get_anno(parens, MetaBC, false) of
```

Comment:
I wonder if we need to check for comments in here. How do they interact with this flipping?

Reply:
I'll add some more tests, good idea

Reply:
Testing for comments was a great idea, it definitely looks better than without checking.
You can see the result in two tests I added at the bottom.
Not sure if it is ideal, but it could be acceptable.